Realm/RLMSchema.mm

```diff
@@ -126,12 +126,20 @@ - (instancetype)init {
 
 - (NSArray *)objectSchema {
     if (!_objectSchema) {
+        if (self == s_sharedSchema) {
+            // Ensure the schema is fully initialized
+            [RLMSchema sharedSchema];
```

Comment:
I wouldn't be opposed to adding a private `partialObjectSchema` property to the type, but I don't think we should pretend that it is the full object schema.